pkg/convert/tf.go

```diff
@@ -1163,16 +1226,42 @@ func convertTemplateExpr(state *convertState,
 				// Other values can be written as is
 				tokens = append(tokens, hclwrite.TokensForValue(lit.Val)...)
 			}
+		}  else if forToken, ok := part.(*hclsyntax.TemplateJoinExpr); ok{
+			// A template join is more complex and contains template control
+			// sections, so delegate to the inner expression.
+			tokens = append(tokens, convertTemplateJoinExpr(state, false, scopes, "", forToken)...)
 		} else {
 			tokens = append(tokens, makeToken(hclsyntax.TokenTemplateInterp, "${"))
 			tokens = append(tokens, convertExpression(state, false, scopes, "", part)...)
 			tokens = append(tokens, makeToken(hclsyntax.TokenTemplateSeqEnd, "}"))
 		}
 	}
-	tokens = append(tokens, makeToken(hclsyntax.TokenCQuote, "\""))
+	if isHeredoc {
+		tokens = append(tokens, makeToken(hclsyntax.TokenCHeredoc, "\nEOT"))
+	} else {
+		tokens = append(tokens, makeToken(hclsyntax.TokenCQuote, "\""))
+	}
 	return tokens
 }
 
+func detectHeredocDelim(state *convertState, r hcl.Range) bool {
+	file := state.sources[r.Filename]
+	start := r.Start.Byte
+	end := start + len("<<EOT")
+	end = min(end, len(file))
+	token := file[start:end]
+
+	if token[0] == '"' {
+		return false
+	}
+
+	if string(token) == "<<EOT" {
```

Comment:
The delimiter in a heredoc can be anything, right? Not sure if TF has some limitations here, but I feel like you could do e.g.

```
<<FOO
...
FOO
```

It's just at least two `<`s (possibly `<<-` is also supported for slurping)

Reply:
Indeed, https://developer.hashicorp.com/terraform/language/expressions/strings#heredoc-strings

Reply:
I added a fancy regex and return delimiters now